### ISSUE TITLE
Enhance Dictionary bulk load for sorted cache

### ISSUE THREAD

**Issue by VrajPandya** [heavyai/heavydb#141]
In method [getOrAddBulk](https://github.com/mapd/mapd-core/blob/617b6425ab5c3e95a316e9d359cddee286be91cf/StringDictionary/StringDictionary.cpp#L242)  We can initially sort the incoming string data and create a temporary vector and merge it with existing sorted cache. This will reduce memmove operations in getOrAdd method for sorted cache.

**Comment by dwayneberry** [comment]
@VrajPandya if you do any work over this area ideally please work on top of my exisiting 
branch

**Comment by psilvest** [comment]
This issue has been inactive for a while, and the project has changed substantially. If this still occurs on the latest release, please add reproduction steps and environment details.

**Comment by VrajPandya** [comment]
Agreed, this is non essential unless specific use case demands the speed up and we can prove that hardware can support it.
